forum/notifications/signals.py

```diff
@@ -1,66 +1,85 @@
 from django.db.models.signals import m2m_changed, post_save
 from django.dispatch import receiver
-from .models import NotificationCategory
+from .models import NotificationCategory, NotificationPreference
 from profiles.models import InvestorProfile, StartupProfile
-from .serializers import StartUpNotificationCreateSerializer
+from .serializers import (
+    StartUpNotificationCreateSerializer,
+    InvestorNotificationCreateSerializer
+)
 from projects.models import Project
 import logging
 
 logger = logging.getLogger(__name__)
 
+def have_preference(notification_receiver, notification_category):
+    startup_preference = NotificationPreference.objects.get(user=notification_receiver.user)
+    if notification_category not in startup_preference.allowed_notification_categories.all():
+        logger.info(f"Startup {notification_receiver.id} does not have a preference for "
+                    f"the '{notification_category.name}' category. Skipping notification creation.")
+        return False
+    return True
+
```

Comment:
Unified call handling and logging

You are repeating the code for checking preferences (have_preference) and creating notifications. This can be moved to a separate function to reduce duplication.